### DISCUSSION TITLE
Add MCNP syntax highlighting for github and sphinx

### DISCUSSION THREAD

**Discussion by MicahGale** [idaholab/MontePy#586] [Ideas]
It would be nice if we could define MCNP syntax in such a way that sphinx, GitHub, etc can understand it. Not sure how difficult this would be.

**Comment by tjlaboss** [comment]
I looked into this for GitHub a while back, and [the answer was a resounding "No."](https://github.com/github-linguist/linguist/issues/2598)

It can be done in Sphinx using [Pygments](https://pygments.org/docs/lexerdevelopment/) by writing your own Lexer. Can it simply import the lexer from MontePy?

**Reply by MicahGale** [reply]
Oh interesting. I bet they are not pluggable, but there could be something hacky to brute force our lexers in there and just ignore `pygments`.

**Reply by MicahGale** [reply]
Oh `pygments` has state switching. That will be helpful. I think this wouldn't be super labor intensive to just create ab-initio.
